src/rules/arrayTypeRule.ts

```diff
@@ -0,0 +1,76 @@
+import * as ts from "typescript";
+
+import * as Lint from "../lint";
+
+const OPTION_ARRAY = "array";
+const OPTION_GENERIC = "generic";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "array-type",
+        description: "Requires using either 'T[]' or 'Array<T>' for arrays.",
+        optionsDescription: Lint.Utils.dedent`
+            One of the following arguments must be provided:
+
+            * \`"${OPTION_ARRAY}"\` enforces use of 'T[]'.
+            * \`"${OPTION_GENERIC}"\` enforces use of 'Array<T>'.`,
+        options: {
+            type: "string",
+            enum: [OPTION_ARRAY, OPTION_GENERIC],
+        },
+        optionExamples: ["[true, array]", "[true, generic]"],
```

Comment:
You can use the `OPTION_` variables here as well (no need to make a commit just to change this though)